### PR TITLE
clientupdate: do not recursively delete dirs in cleanupOldDownloads

### PR DESCRIPTION
**Pull request by awly**
In case there's a wild symlink in one of the target paths, we don't want to accidentally delete too much. Limit `cleanupOldDownloads` to deleting individual files only.

Updates https://github.com/tailscale/tailscale/issues/10082

### CODE REVIEW

File: clientupdate/clientupdate.go [hunk 1]
```diff
@@ -833,14 +833,17 @@ func (up *Updater) installMSI(msi string) error {
 	return err
 }
 
+// cleanupOldDownloads removes all files matching glob (see filepath.Glob).
+// Only regular files are removed, so the glob must match specific files and
```
Comment (bradfitz) [inline clientupdate/clientupdate.go:837]: You didn't check they're regular.

Comment (awly) [reply clientupdate/clientupdate.go:837]: Added the check

File: clientupdate/clientupdate.go [hunk 2]
```diff
@@ -833,14 +833,25 @@ func (up *Updater) installMSI(msi string) error {
...
+// not directories.
 func (up *Updater) cleanupOldDownloads(glob string) {
 	matches, err := filepath.Glob(glob)
 	if err != nil {
 		up.Logf("cleaning up old downloads: %v", err)
 		return
 	}
 	for _, m := range matches {
-		if err := os.RemoveAll(m); err != nil {
+		s, err := os.Stat(m)
```
Comment (tylersmalley) [inline clientupdate/clientupdate.go:846]: I think we want LStat, so it doesn't follow a symlink

Comment (awly) [reply clientupdate/clientupdate.go]: :facepalm:

**Review by tylersmalley — APPROVED** [review]

**Review by bradfitz — APPROVED** [review]

**Pull request merged**